client/state/posts/revisions/reducer.js

```diff
@@ -42,13 +38,11 @@ export function requesting( state = {}, action ) {
 export function revisions( state = {}, action ) {
 	if ( action.type === POST_REVISIONS_RECEIVE ) {
 		const { siteId, postId } = action;
-		return {
-			...state,
+		return merge( {}, state, {
```

Comment:
nice catch! 

> Sorry, If I confused you with my previous comment, I like to use merge in reducer of scalar values (like above) but not when the reducers objects with unknown shapes.

hehe np, that's why code reviews are great :) that makes a lot of sense, will keep it in mind :) 

I've added a "test" as well as reverting back to using spreads for that reducer.